src/op_analytics/datapipeline/models/templates/refined_transactions_fees.sql.j2

```diff
@@ -0,0 +1,147 @@
+    t.from_address,
+    t.to_address,
+    t.block_number,
+    t.block_timestamp,
+    t.hash,
+    t.gas_price,
+    t.gas AS gas_limit,
+    t.receipt_gas_used,
+    t.receipt_l1_gas_used,
+    -- Fees
+    t.receipt_l1_gas_price,
+    t.receipt_l1_blob_base_fee,
+    b.base_fee_per_gas,
+    t.max_priority_fee_per_gas,
+    t.transaction_type,
+    CASE WHEN t.gas_price = 0 THEN 0 ELSE t.receipt_l1_fee END AS l1_fee,
+    -- L2 Fees and breakdown into BASE and PRIORITY contributions
+    if(t.max_priority_fee_per_gas = 0, t.gas_price - b.base_fee_per_gas, 0) AS legacy_extra_fee_per_gas,
+    CASE WHEN t.gas_price = 0 THEN 0 ELSE t.gas_price * t.receipt_gas_used END AS l2_fee,
+    CASE WHEN t.gas_price = 0 THEN 0 ELSE t.max_priority_fee_per_gas * t.receipt_gas_used END AS l2_priority_fee,
+    -- Convenience columns
+    CASE WHEN t.gas_price = 0 THEN 0 ELSE b.base_fee_per_gas * t.receipt_gas_used END AS l2_base_fee,
+    epoch_to_hour(t.block_timestamp) AS block_hour,
+    hexstr_method_id(t.input) AS method_id,
+    -- L1 Fee BASE and BLOB scalars
+    t.receipt_status = 1 AS success,
+    micro(t.receipt_l1_base_fee_scalar) AS l1_base_fee_scalar,
+    t.receipt_l1_fee_scalar::DECIMAL(12, 6) AS l1_fee_scalar,
+
+    -- Transaction Attributes
+    coalesce(micro(t.receipt_l1_blob_base_fee_scalar), 0) AS l1_blob_base_fee_scalar,
+    hexstr_bytelen(t.input) AS input_byte_length,
+    hexstr_zero_bytes(t.input) AS input_zero_bytes,
+    hexstr_nonzero_bytes(t.input) AS input_nonzero_bytes,
+    t.transaction_type = 126 AS is_system_transaction,
+    ( -- Attributes Deposited Spec https://specs.optimism.io/protocol/deposits.html#l1-attributes-deposited-transaction
+      t.transaction_type = 126
+      AND t.from_address = lower('0xDeaDDEaDDeAdDeAdDEAdDEaddeAddEAdDEAd0001')
+      AND t.to_address = lower('0x4200000000000000000000000000000000000015')
+    ) AS is_attributes_deposited_transaction
+
+  FROM ingestion_transactions_v1 AS t
+  INNER JOIN pb AS b
+    ON t.block_number = b.number
+  WHERE 1 = 1
+-- Optional address filter for faster results when developing.
+-- AND from_address LIKE '0x00%'  
+),
+
+
+-- Add fallback l1 gas estimate based on observed fee and weighted gas and blob prices.
+pt2 AS (
+  SELECT
+    *,
+    coalesce(
+      receipt_l1_gas_used,
+      round(16 * estimated_size)::INT64
+    ) AS l1_gas_used
+  FROM (
+    SELECT
+      *,
+      -- Estimated Size in bytes calculation, post-Fjord. This is not equivalent to L1 Gas Used
+      l1_fee / (
+        (16 * l1_base_fee_scalar * receipt_l1_gas_price)
+        + (l1_blob_base_fee_scalar * receipt_l1_blob_base_fee)
+      ) AS estimated_size_bytes
+    FROM pt1
+  )
+),
+
+-- Add more fee calculations.
+-- Reuses results from the previous CTEs.
+
+pt3 AS (
+  SELECT
+    *,
+    --
+    -- Total fee
+    l2_fee + l1_fee AS tx_fee,
+    -- 
+    -- L2 Legacy Fee
+    if(l2_priority_fee = 0, l2_fee - l2_base_fee, 0) AS l2_legacy_extra_fee,
+    -- 
+    -- L1 Base 
```

Comment:
We should build tests here that l1_base_fee + l1_blob_fee = l1_fee

This should hold through the history of time.